src/main/java/com/zombiecastlerush/util/Prompter.java

```diff
@@ -61,10 +65,12 @@ static void advanceGame(Player player) throws JsonProcessingException {
                         System.out.println(player.displayStatus());
                     break;
                 case "pick-up":
-                    for (Item item : currentRoom.getInventory().getItems()) {
-                        if (item.getName().equalsIgnoreCase(userInputList.get(1))) {
-                            player.pickUp(item);
-                            break;
+                    if (!(currentRoom instanceof Shop)) {
```

Comment:
All the casting and the size of this switch case makes me uneasy, but I don't know how to fix it without a very involved refactoring... which is beyond the scope of this PR.